scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -518,6 +522,9 @@ def add_testsuites(self, testsuite_filter=None):
                         )
                         suite_yaml_path = alt_config
                         break
+                testsuite_patterns_r = []
+                for pattern in testsuite_pattern:
+                    testsuite_patterns_r.append(re.compile(pattern))
```

Comment:
can be moved to the top ot that method, to not repead regex compilation when iterating through yaml files.